src/pytorch_lightning/trainer/trainer.py

```diff
@@ -644,12 +644,16 @@ def _call_and_handle_interrupt(self, trainer_fn: Callable, *args: Any, **kwargs:
             if not self.interrupted:
                 self.state.status = TrainerStatus.INTERRUPTED
                 self._call_callback_hooks("on_exception", exception)
+                for logger in self.loggers:
+                    logger.finalize("failed")
```

Comment:
Right now, KeyboardInterrupt is the same as "failed". Perhaps we want to introduce something like "stopped" in the future. How the loggers react here depends on how the third-party implementations expect things to be reported when an experiment gets "stopped" or "interrupted" in a controlled way.